### PR TITLE
DRV-216: Support for versioned @query type

### PR DESCRIPTION
**Pull request by trevorsibanda**

### CODE REVIEW

File: faunadb/objects.py [hunk 1]
```diff
@@ -170,6 +171,16 @@ class Query(_Expr):
...
   """
 
+  def __init__(self, value):
+    """
+    :param value:
+      If an api_version key is not defined, `faunadb.__api_version__` is used.
+    """
+    if isinstance(value, dict):
+      if not "api_version" in value:
+        value["api_version"] =  __api_version__
```
Comment (marrony) [inline faunadb/objects.py:181]: please don't do this, we should not use the default api version (`v3`) case it was not provided, ideally, users shouldn't be constructing query types

File: tests/test_query.py [hunk 2]
```diff
@@ -60,6 +60,14 @@ def _assert_bad_query(self, q):
   def test_abort(self):
     self._assert_bad_query(query.abort("aborting"))
 
+  def test_query(self):
```
Comment (marrony) [inline tests/test_query.py:63]: I can only see two basic tests here:
1. create a `Query` instance (with no `api_version` set) and send it to server, and asset that the echoed value contains `api_version: "2.12"`
2. call the function `query()` and assert the returned value contains `api_version: "3"`

File: tests/test_serialization.py [hunk 3]
```diff
@@ -44,8 +44,10 @@ def test_bytes(self):
       self.assertJson(b'\x01\x02\x03', '{"@bytes":"AQID"}')
 
   def test_query(self):
-    self.assertJson(Query({"lambda": "x", "expr": {"var": "x"}}),
-                    '{"@query":{"expr":{"var":"x"},"lambda":"x"}}')
+    expected1 = '{"@query":{"expr":{"var":"x"},"lambda":"x"}}'
+    self.assertJson(Query({"lambda": "x", "expr": {"var": "x"}}), expected1)
+    self.assertJson(Query({"api_version": "3", "lambda": "x", "expr": {"var": "x"}}),
+                    '{"@query":{"api_version":"3","expr":{"var":"x"},"lambda":"x"}}')
```
Comment (evbruno) [inline tests/test_serialization.py:50]: a minor thing here: are you going to extract the expected string to `expected2`? or inline both asserts?  (I don't mind, just asking for consistency)

File: tests/test_query.py (hunk 2 above)
Comment (evbruno) [reply tests/test_query.py:63]: I `unresolved` that comment: I'm not seeing the two tests here. Am I missing something?

**Pull request merged**